From f96788c66542de33a08ed10b0383ad5e44b375d4 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Sun, 6 Aug 2017 00:06:56 -0400
Subject: [PATCH] Replace 'redhat-release' with generic 'system-release'

fedora-release and mageia-release offer generic 'system-release'
Provides that can be used to determine the release version of the
distribution. Derivatives of Fedora (including RHEL) also offer this
generic capability in their distribution release packages.
---
 examples/solv/repoinfo_config_yum.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/examples/solv/repoinfo_config_yum.c b/examples/solv/repoinfo_config_yum.c
index 697bd6d..efccf1e 100644
--- a/examples/solv/repoinfo_config_yum.c
+++ b/examples/solv/repoinfo_config_yum.c
@@ -48,7 +48,7 @@ yum_substitute(Pool *pool, char *line)
 	
 	      queue_init(&q);
 	      rpmstate = rpm_state_create(pool, pool_get_rootdir(pool));
-	      rpm_installedrpmdbids(rpmstate, "Providename", "redhat-release", &q);
+	      rpm_installedrpmdbids(rpmstate, "Providename", "system-release", &q);
 	      if (q.count)
 		{
 		  void *handle;
@@ -62,7 +62,7 @@ yum_substitute(Pool *pool, char *line)
 	      queue_free(&q);
 	      if (!releaseevr)
 		{
-		  fprintf(stderr, "no installed package provides 'redhat-release', cannot determine $releasever\n");
+		  fprintf(stderr, "no installed package provides 'system-release', cannot determine $releasever\n");
 		  exit(1);
 		}
 	    }
-- 
2.14.1

